### ISSUE TITLE
Make storage work

### ISSUE THREAD

**Issue by mismosmi** [farbenmeer/editor#8]
For images we need some generic way to upload to and view images from S3-compatible storage.

**Comment by garciamarin** [comment]
Can we maybe do this with signed urls? 

We would just need 2 async functions provided at some point by the user

```ts
async function getSignedGetUrl(fileKey: string): string { return signedUrl }
async function getSignedPostUrl(fileKey: string): PresignedPost { return post}
```
and a `bucketPostUrl`. To upload we would do something like

```ts
const { url, fields } = await fetch(  getSignedPostUrl((fileKey) )
await fetch(url, {
    method: 'POST',
    body: formData(...fields...) ,
  })
```
and similarly for the fetching images from the bucket. 

Would something like this workout?

**Comment by mismosmi** [comment]
That should work We can probably just throw some server actions in there.

We should probably reimplement something like https://github.com/farbenmeer/handbuch/blob/main/src/components/S3Image.tsx here.

**Comment by garciamarin** [comment]
yes! like that looks verygood! Btw its very similar to how we do it too ^^

What I dont quite get, is when we get this info from the user (functions and what not). Is it something like: 
user installs package -> 
user creates "instance" of desired type of RTE and if necessary provides the needed functions -> 
User can use the RTE and add photos by giving props to the `<RTE/>` component?

**Comment by mismosmi** [comment]
I'd expect something like
```ts
<Editor getSignedImageUrl={...} getSignedUploadUrl={...} plugins={...} getMentions={...} />
```
I guess 🤷

**Comment by mismosmi** [comment]
Or maybe
```tsx
<Editor plugins={[imagePlugin({ getSignedImageUrl, getSignedUploadUrl })]} />
```

**Comment by garciamarin** [comment]
alright I see.!

We can always check at the end if it became annoying to have to pass in many porps everytime when one uses the RTE or if its actually ok. 

The second options looks nicer to me!

**Comment by mismosmi** [comment]
if it gets annoying you can always pull out the plugin initialization
```ts
const plugins = [imagePlugin(...), mentionPlugin(...)]
```
and then use that wherever you use the editor
```tsx
<Editor plugins={plugins} />
```
or even wrap the editor
```tsx
function MyEditor() {
  return <Editor {...} />
}
```

**Comment by mismosmi** [comment]
I like passing parameters to the plugin that needs them better as well.
